token-metadata/program/src/instruction.rs

```diff
@@ -0,0 +1,120 @@
+use {
+    borsh::{BorshDeserialize, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        sysvar,
+    },
+    std::str::FromStr,
+};
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+/// Args for Create call
+pub struct CreateMetadataAccountArgs {
+    /// The name of the asset
+    pub name: String,
+    /// The symbol for the asset, ie, AAPL or SHOES
+    pub symbol: String,
+    /// URI pointing to JSON representing the asset
```

Comment:
sorry to nit but this is a URL if it points at something, not a URI

Reply:
I would prefer to keep it URI -> URI is more generic - it can point to more things, as it's a superset of URL, and it keeps backward compatibility with ERC-721.

Reply:
Ok, then let's drop the `pointing to JSON representing the asset` part because the URI here there might not be a URL to a JSON file.

Aside: what's the thinking on where the documentation for the JSON file convention will live?

Reply:
Sure, we can drop that comment. Current plan is to follow ERC-1155 metadata json format, that being said it is an extension that is not really part of this PR. (https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md#erc-1155-metadata-uri-json-schema)

Reply:
Going the other way and just saying field is a `eip_1155_metadata_uri: String` would work too.  EIP 1155 has everything we need right?  I'm not bashful about reusing well-known existing standards

Reply:
The field name shouldn't contain the standard. It's a URI, technically someone can store something else in it, and it's the client's responsibility to be able to decode different things.

Reply:
What else do we envision this field holding though, if not a link to EIP 1155-compatible metadata?

Reply:
ERC-20 doesn't specify what it'd be, so it could be a json, could be a png, etc.

Reply:
I'm worried that just throwing our hands up and saying, "it's a link to something" just pushes the problem to each individual wallet/integration who will all do their own slightly incompatible thing if we don't provide them more guidance.